src/query/graph.cpp

```diff
@@ -36,6 +39,18 @@ void Graph::Expand(const Path &path) {
   std::for_each(path_edges_.begin(), path_edges_.end(), [this](const EdgeAccessor e) { edges_.insert(e); });
 }
 
+void Graph::Expand(TypedValue::TVector const &nodes, TypedValue::TVector const &edges) {
+  auto actual_nodes = nodes | rv::filter([](auto const &each) { return each.type() == TypedValue::Type::Vertex; }) |
+                      rv::transform([](auto const &each) { return each.ValueVertex(); });
+  vertices_.insert(actual_nodes.begin(), actual_nodes.end());
+
+  auto actual_edges =
+      edges | rv::filter([](auto const &each) { return each.type() == TypedValue::Type::Edge; }) |
+      rv::transform([](auto const &each) { return each.ValueEdge(); }) |
+      rv::filter([this](auto const &each) { return vertices_.contains(each.From()) && vertices_.contains(each.To()); });
```

Comment:
Is this the intended behaviour, constrained to the subgraph induced by the vertices?
An alternative is to add the missing vertices...just double check what is desired functionality and ensure it is documented.